### PR TITLE
Arrow keys block bug and remove focus plugin

### PR DESCRIPTION
**Pull request by efjacobson**
i made some small changes to https://github.com/triniti/cms-js/tree/arrow-keys-block-bug and then removed the focus plugin and here we are. the focus plugin stopped working a while back and no one noticed so clearly its not important

@nanetteCodes can you make sure i didnt mess up any of your stuff?
@arunkarnati can you review the removal of the focus plugin?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 * Add `aspect_ratio` ui for gallery and document blocks
 * Use `@triniti/schemas` v1.1.6
 * Create `VideoAssetPickerField` and use for video `mezzanine_ref`.
+* Up/Down arrow key in blocks bug
```
Comment (arunkarnati) [inline CHANGELOG.md]: @nanetteCodes - please describe briefly what the bug with the `up/down`  is. Right now, it is unclear to me.

Comment (efjacobson) [reply CHANGELOG.md]: @nanetteCodes put the name of the ticket

File: src/plugins/blocksmith/components/blocksmith/index.jsx [hunk 2]
```diff
@@ -844,67 +827,19 @@ class Blocksmith extends React.Component {
...
    * @returns {string} 'handled' (meaning we did something special) or 'not-handled' (meaning we
    * want to allow the editor to take over and resume default behavior).
    */
   handleKeyCommand(command) {
-    const { activeBlockKey, editorState } = this.state;
-    let anchorKey;
-    let currentBlock;
-    let newBlockKey;
+    const { editorState } = this.state;
     let newContentState;
```
Comment (arunkarnati) [inline src/plugins/blocksmith/components/blocksmith/index.jsx]: @efjacobson - quick suggestion here: if you'd like to declare variables that are used in a specific switch case clause and not be available to the entire switch block, then you can wrap that clause in curly brackets. Currently, the three variables are needed only for the first case, so it keeps clean to have them just inside that one case.

Comment (efjacobson) [reply src/plugins/blocksmith/components/blocksmith/index.jsx]: sweet, did not know that. updated

File: src/plugins/blocksmith/components/blocksmith/index.jsx [hunk 3]
```diff
@@ -992,53 +901,57 @@ class Blocksmith extends React.Component {
...
-          if (nextBlock) {
-            this.setState({
-              editorState: selectBlock(editorState, nextBlock),
-            });
-          }
+          e.preventDefault(); // would be going "into" an atomic block
+        }
+        break;
+      case 'ArrowDown':
+        areRestOfBlocksAtomic = !blocksAsArray.slice(currentBlockIndex + 1, blocksAsArray.length).find((b) => b.getType() !== 'atomic');
```
Comment (arunkarnati) [inline src/plugins/blocksmith/components/blocksmith/index.jsx]: Whoa! 👍

File: src/plugins/blocksmith/resolver.js [hunk 4]
```diff
@@ -198,7 +198,7 @@ export const getModalComponent = (message) => {
  *
  * @returns {?Component} a React component that is intended to go inside the DraftJs editor
  */
-export const getPlaceholder = (type, decorator, props = {}) => {
+export const getPlaceholder = (type, props = {}) => {
```
Comment (arunkarnati) [inline src/plugins/blocksmith/resolver.js]: @efjacobson - please update the DocBlock as well.

File: src/plugins/blocksmith/utils/getLines.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import DraftOffsetKey from 'draft-js/lib/DraftOffsetKey';
+
+/**
+ * Calculates and returns an array of the individual lines of a text block. Useful for
+ * some keyboard navigation behavior.
+ *
+ * @param {EditorState} editorState - a state instance of a DraftJs Editor
+ *
+ * @returns {boolean}
```
Comment (arunkarnati) [inline src/plugins/blocksmith/utils/getLines.js]: @efjacobson - please change the return type to `Array`

File: src/plugins/blocksmith/resolver.js (hunk 4 above)
Comment (efjacobson) [reply src/plugins/blocksmith/resolver.js]: @arunkarnati done

File: src/plugins/blocksmith/utils/getLines.js (hunk 5 above)
Comment (efjacobson) [reply src/plugins/blocksmith/utils/getLines.js]: @arunkarnati done

File: CHANGELOG.md (hunk 1 above)
Comment (nanetteCodes) [reply CHANGELOG.md]: @arunkarnati / @efjacobson I updated the changelog

**Comment by efjacobson** [comment]
diff is suspiciously weird on this, closing in favor of https://github.com/triniti/cms-js/pull/190

**Pull request closed without merging**